src/min_heap.rs

```diff
@@ -0,0 +1,210 @@
+use crate::base_vec::{BaseVec, InitError};
+use crate::storable::BoundedStorable;
+use crate::{GrowFailed, Memory};
+use std::fmt;
+
+#[cfg(test)]
+mod tests;
+
+const MAGIC: [u8; 3] = *b"SMH"; // Short for "stable min heap".
+
+/// An implementation of a binary heap.
+///
+/// Contrary to [std::collections::BinaryHeap], this heap is a min-heap (smallest items come first).
+/// Motivation: max heaps are helpful for sorting, but most daily programming tasks require min
+/// heaps.
+pub struct MinHeap<T: BoundedStorable + PartialOrd, M: Memory>(BaseVec<T, M>);
```

Comment:
nit: shall we add this to the list of available data structures in the [readme](https://github.com/dfinity/stable-structures#available-data-structures)?

Reply:
Done. I've also added a changelog file.